Handle builds without C++

libxmlrpc nicely handles the fact of being built without C++ support,
except for two locations, fixed by this patch.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Alvaro Gamez Machado <alvaro.gamez@hazent.com>

Index: b/lib/util/Makefile
===================================================================
--- a/lib/util/Makefile
+++ b/lib/util/Makefile
@@ -41,11 +41,14 @@
 LIBOBJS = \
   casprintf.o \
   cmdline_parser.o \
-  cmdline_parser_cpp.o \
   getoptx.o \
   string_parser.o \
   stripcaseeq.o \
 
+ifeq ($(ENABLE_CPLUSPLUS),yes)
+LIBOBJS += cmdline_parser_cpp.o
+endif
+
 .PHONY: all
 all: $(LIBOBJS)
 
Index: b/lib/Makefile
===================================================================
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -9,7 +9,10 @@
 
 # Build up SUBDIRS:
 SUBDIRS = 
-SUBDIRS += util libutil libutil++
+SUBDIRS += util libutil
+ifeq ($(ENABLE_CPLUSPLUS),yes)
+  SUBDIRS += libutil++
+endif
 ifeq ($(ENABLE_ABYSS_SERVER),yes)
   SUBDIRS += abyss
   ifeq ($(ENABLE_CPLUSPLUS),yes)
